File: core/src/com/t2wonderland/kurona/screens/SelectScreen.java
package com.t2wonderland.kurona.screens;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.Screen;
import com.badlogic.gdx.graphics.GL20;
import com.badlogic.gdx.graphics.OrthographicCamera;
import com.badlogic.gdx.math.Rectangle;
import com.badlogic.gdx.math.Vector3;
import com.t2wonderland.kurona.Assets;
import com.t2wonderland.kurona.KuronanDash;
import com.t2wonderland.kurona.models.OverlapCheker;

public class SelectScreen implements Screen {

	final KuronanDash game;
	OrthographicCamera camera;
	Rectangle kuronaBounds;
	Rectangle komaBounds;
	Rectangle shishimaruBounds;
	Rectangle backBounds;
	Rectangle startBounds;

	public SelectScreen(final KuronanDash gam){
		game = gam;
		camera = new OrthographicCamera(KuronanDash.displayWidth, KuronanDash.displayHeight);
		camera.setToOrtho(false, KuronanDash.displayWidth, KuronanDash.displayHeight);
		
		// regions of the buttons
		kuronaBounds = Assets.selectKurona.getBoundingRectangle();
		komaBounds = Assets.selectKoma.getBoundingRectangle();
		shishimaruBounds = Assets.selectShishimaru.getBoundingRectangle();
		backBounds = Assets.selectBack.getBoundingRectangle();
		startBounds = Assets.selectStart.getBoundingRectangle();
	}

	public void update (float deltaTime) {
		if (Gdx.input.isTouched(0)) {
			Vector3 touchPos = new Vector3();
			touchPos.set(Gdx.input.getX(), Gdx.input.getY(), 0);
			camera.unproject(touchPos);

			if (OverlapCheker.pointInRectangle(kuronaBounds, touchPos.x, touchPos.y)) {
				Assets.playSound(Assets.clickSound);
				return;
			}
			if (OverlapCheker.pointInRectangle(komaBounds, touchPos.x, touchPos.y)) {
				Assets.playSound(Assets.clickSound);
				return;
			}
			if (OverlapCheker.pointInRectangle(shishimaruBounds, touchPos.x, touchPos.y)) {
				Assets.playSound(Assets.clickSound);
				return;
			}
			if (OverlapCheker.pointInRectangle(backBounds, touchPos.x, touchPos.y)) {
				Assets.playSound(Assets.clickSound);
				game.setScreen(new TitleScreen(game));
				return;
			}
			if (OverlapCheker.pointInRectangle(startBounds, touchPos.x, touchPos.y)) {
				Assets.playSound(Assets.clickSound);
				game.setScreen(new GameScreen(game));
				return;
			}
		}
	}
	
	public void draw (float deltaTime) {
		Gdx.gl.glClearColor(0, 0.3f, 1, 1);
		Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT);
		camera.update();
        game.batch.setProjectionMatrix(camera.combined);
        game.batch.begin();
        Assets.selectKurona.draw(game.batch);
        Assets.selectKoma.draw(game.batch);
        Assets.selectShishimaru.draw(game.batch);
        Assets.selectDescription.draw(game.batch);
        Assets.selectBack.draw(game.batch);
        Assets.selectStart.draw(game.batch);
		game.batch.end();
	}
	
	@Override
	public void render(float delta) {
		update(delta);
		draw(delta);
	}

	@Override
	public void resize(int width, int height) {
	}

	@Override
	public void show() {
	}

	@Override
	public void hide() {
	}

	@Override
	public void pause() {
	}

	@Override
	public void resume() {
	}

	@Override
	public void dispose() {
	}

}

File: core/src/com/t2wonderland/kurona/Screens/SelectScreen.kt
package com.t2wonderland.kurona.Screens

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.Screen
import com.badlogic.gdx.graphics.GL20
import com.badlogic.gdx.graphics.OrthographicCamera
import com.badlogic.gdx.math.Rectangle
import com.badlogic.gdx.math.Vector3
import com.t2wonderland.kurona.Assets
import com.t2wonderland.kurona.KuronanDash
import com.t2wonderland.kurona.Models.CharacterSelect
import com.t2wonderland.kurona.Models.OverlapChecker

class SelectScreen(internal val game: KuronanDash) : Screen {
    internal var camera: OrthographicCamera
    internal var boundsKurona: Rectangle
    internal var boundsKoma: Rectangle
    internal var boundsShishimaru: Rectangle
    internal var boundsBack: Rectangle
    internal var boundsStart: Rectangle
    internal var selectedCharacter: CharacterSelect

    init {
        camera = OrthographicCamera(KuronanDash.displayWidth.toFloat(), KuronanDash.displayHeight.toFloat())
        camera.setToOrtho(false, KuronanDash.displayWidth.toFloat(), KuronanDash.displayHeight.toFloat())

        // regions of the buttons
        boundsKurona = Assets.selectKurona.boundingRectangle
        boundsKoma = Assets.selectKoma.boundingRectangle
        boundsShishimaru = Assets.selectShishimaru.boundingRectangle
        boundsBack = Assets.selectBack.boundingRectangle
        boundsStart = Assets.selectStart.boundingRectangle

        selectedCharacter = CharacterSelect.Kurona
    }

    fun update(deltaTime: Float) {
        if (Gdx.input.isTouched(0)) {
            val touchPos = Vector3()
            touchPos.set(Gdx.input.x.toFloat(), Gdx.input.y.toFloat(), 0f)
            camera.unproject(touchPos)

            if (OverlapChecker.pointInRectangle(boundsKurona, touchPos.x, touchPos.y)) {
                Assets.playSound(Assets.clickSound)
                selectedCharacter = CharacterSelect.Kurona
                return
            }
            if (OverlapChecker.pointInRectangle(boundsKoma, touchPos.x, touchPos.y)) {
                Assets.playSound(Assets.clickSound)
                selectedCharacter = CharacterSelect.Koma
                return
            }
            if (OverlapChecker.pointInRectangle(boundsShishimaru, touchPos.x, touchPos.y)) {
                Assets.playSound(Assets.clickSound)
                selectedCharacter = CharacterSelect.Shishimaru
                return
            }
            if (OverlapChecker.pointInRectangle(boundsBack, touchPos.x, touchPos.y)) {
                Assets.playSound(Assets.clickSound)
                game.screen = TitleScreen(game)
                return
            }
            if (OverlapChecker.pointInRectangle(boundsStart, touchPos.x, touchPos.y)) {
                Assets.playSound(Assets.clickSound)
                game.screen = GameScreen(game, selectedCharacter)
                return
            }
        }
    }

    fun draw(deltaTime: Float) {
        Gdx.gl.glClearColor(0f, 0.3f, 1f, 1f)
        Gdx.gl.glClear(GL20.GL_COLOR_BUFFER_BIT)
        camera.update()
        game.batch.projectionMatrix = camera.combined

        game.batch.begin()
        Assets.selectKurona.draw(game.batch)
        Assets.selectKoma.draw(game.batch)
        Assets.selectShishimaru.draw(game.batch)
        Assets.selectDescription.draw(game.batch)
        Assets.selectBack.draw(game.batch)
        Assets.selectStart.draw(game.batch)
        game.batch.end()
    }

    override fun render(delta: Float) {
        update(delta)
        draw(delta)
    }

    override fun resize(width: Int, height: Int) {
    }

    override fun show() {
    }

    override fun hide() {
    }

    override fun pause() {
    }

    override fun resume() {
    }

    override fun dispose() {
    }

}
